tools/clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -3333,6 +3333,12 @@ EmitConditionalOperatorLValue(const AbstractConditionalOperator *expr) {
 /// cast from scalar to union.
 LValue CodeGenFunction::EmitCastLValue(const CastExpr *E) {
   // HLSL Change Begins
+  if (E->getCastKind() == CK_HLSLRowMajorToColMajor ||
+      E->getCastKind() == CK_HLSLColMajorToRowMajor) {
+    // Orientation cast on LValue.
+    // Do the cast when copyin/out by create LValue with target type.
+    return EmitLValue(E->getSubExpr());
```

Comment:
This seems like a problem. If the orientation impacts the data layout, we can't create an l-value cast without transposing going in _and_ back out of the l-value because any writes need to be transposed.